src/software/ai/motion_constraint/motion_constraint_visitor.h

```diff
@@ -50,14 +39,17 @@ class MotionConstraintManager : public TacticVisitor
     void visit(const StopTestTactic &tactic) override;
     void visit(const GoalieTestTactic &tactic) override;
 
-   private:
-    std::set<MotionConstraint> current_allowed_constraints;
 
     /**
-     * Adds move constraints determined from gamestate to current_motion_constraints
+     * Gets the current allowed constraints from a tactic
      *
-     * @param game_state GameState to generate move constraints from
+     * @param The tactic to register
+     *
+     * @modifies current_allowed_constraints
+     * @return set of MotionConstraints
      */
-    std::set<MotionConstraint> getMotionConstraintsFromGameState(
-        const GameState &game_state) const;
+    std::set<MotionConstraint> getCurrentAllowedConstraints(Tactic &tactic);
```

Comment:
```suggestion
    std::set<MotionConstraint> getCurrentAllowedConstraints(const Tactic &tactic);
```
similarly, this can be const with the changes to tactic visitor

Reply:
Hmm, I ran into a problem adding the const. If I add const to the tactic, I will also have to add const to the accept function, which means that I have to add the const to all of the tactics since they all inherit the acception function from tactics.h. Should I do this?

Reply:
yes that would be good